### PR TITLE
Run clang-format on src/runtime

### PR DESCRIPTION
**Pull request by steven-johnson**
Apparently the C++ code in src/runtime missed out on the Clang Format Party; applying it all in one go here.

### CODE REVIEW

File: src/runtime/HalideBuffer.h [hunk 1]
```diff
@@ -437,7 +440,8 @@ class Buffer {
         buf.host = old_buf.host;
         buf.type = t;
         int d;
-        for (d = 0; d < 4 && old_buf.extent[d]; d++);
+        for (d = 0; d < 4 && old_buf.extent[d]; d++)
```
Comment (abadams) [inline src/runtime/HalideBuffer.h]: A bit weird. Maybe the whole thing should be:

```
int d = 0;
while (d < 4 && old_buf.extent[d]) {
    d++;
}
```

File: src/runtime/cl_functions.h [hunk 2]
```diff
@@ -213,140 +128,36 @@ CL_FN(cl_int,
...
-      clEnqueueNDRangeKernel, (cl_command_queue /* command_queue */,
-                               cl_kernel        /* kernel */,
-                               cl_uint          /* work_dim */,
-                               const size_t *   /* global_work_offset */,
-                               const size_t *   /* global_work_size */,
-                               const size_t *   /* local_work_size */,
-                               cl_uint          /* num_events_in_wait_list */,
-                               const cl_event * /* event_wait_list */,
-                               cl_event *       /* event */));
+      clEnqueueMapImage, (cl_command_queue /* command_queue */, cl_mem /* image */, cl_bool /* blocking_map */, cl_map_flags /* map_flags */, const size_t * /* origin[3] */, const size_t * /* region[3] */, size_t * /* image_row_pitch */, size_t * /* image_slice_pitch */, cl_uint /* num_events_in_wait_list */, const cl_event * /* event_wait_list */, cl_event * /* event */, cl_int * /* errcode_ret */));
```
Comment (abadams) [inline src/runtime/cl_functions.h]: The changes to this file seem like a pessimizaton

File: src/runtime/cuda.cpp [hunk 3]
```diff
@@ -395,11 +408,11 @@ WEAK CUresult create_cuda_context(void *user_context, CUcontext *ctx) {
 
         // threads per core is a function of the compute capability
         int threads_per_core = (cc_major == 1 ? 8 :
-                                cc_major == 2 ? (cc_minor == 0 ? 32 : 48) :
-                                cc_major == 3 ? 192 :
-                                cc_major == 5 ? 128 :
-                                cc_major == 6 ? (cc_minor == 0 ? 64 : 128) :
-                                cc_major == 7 ? 64 : 0);
+                                                cc_major == 2 ? (cc_minor == 0 ? 32 : 48) :
```
Comment (abadams) [inline src/runtime/cuda.cpp]: runaway indentaton

File: src/runtime/cuda.cpp [hunk 4]
```diff
@@ -576,13 +591,15 @@ namespace Internal {
 WEAK halide_device_allocation_pool cuda_allocation_pool;
 
 __attribute__((constructor))
-WEAK void register_cuda_allocation_pool() {
+WEAK void
+register_cuda_allocation_pool() {
```
Comment (abadams) [inline src/runtime/cuda.cpp]: I see this in a few places. Not sure what's going on.

File: src/runtime/profiler.cpp [hunk 5]
```diff
@@ -331,44 +335,50 @@ WEAK void halide_profiler_report_unlocked(void *user_context, halide_profiler_st
 
                 sstr << "  " << fs->name << ": ";
                 cursor += 25;
-                while (sstr.size() < cursor) sstr << " ";
+                while (sstr.size() < cursor)
```
Comment (abadams) [inline src/runtime/profiler.cpp]: while loops in this function now need braces

File: src/runtime/profiler_inlined.cpp [hunk 6]
```diff
@@ -6,26 +6,31 @@ WEAK __attribute__((always_inline)) int halide_profiler_set_current_func(halide_
     // Use empty volatile asm blocks to prevent code motion. Otherwise
     // llvm reorders or elides the stores.
     volatile int *ptr = &(state->current_func);
-    asm volatile ("":::);
+    asm volatile("" ::
```
Comment (abadams) [inline src/runtime/profiler_inlined.cpp]: Interesting... I wonder if clang-format even understands asm statements.

File: src/runtime/tracing.cpp [hunk 7]
```diff
@@ -211,7 +217,8 @@ WEAK int32_t halide_default_trace(void *user_context, const halide_trace_event_t
 
         // Round up bits to 8, 16, 32, or 64
         int print_bits = 8;
-        while (print_bits < e->type.bits) print_bits <<= 1;
+        while (print_bits < e->type.bits)
```
Comment (abadams) [inline src/runtime/tracing.cpp]: braces

File: src/runtime/write_debug_image.cpp [hunk 8]
```diff
@@ -92,8 +91,10 @@ struct halide_tiff_header {
 
 WEAK bool ends_with(const char *filename, const char *suffix) {
     const char *f = filename, *s = suffix;
-    while (*f) f++;
-    while (*s) s++;
+    while (*f)
```
Comment (abadams) [inline src/runtime/write_debug_image.cpp]: braces

File: src/runtime/write_debug_image.cpp [hunk 9]
```diff
@@ -216,24 +219,27 @@ WEAK extern "C" int32_t halide_debug_to_file(void *user_context, const char *fil
...
                 }
             }
         }
     } else if (ends_with(filename, ".mat")) {
         // Construct a name for the array from the filename
         const char *start, *end;
-        for (end = filename; *end; end++);
-        for (; *end != '.'; end--);
-        for (start = end; start != filename && start[-1] != '/'; start--);
+        for (end = filename; *end; end++)
```
Comment (abadams) [inline src/runtime/write_debug_image.cpp]: This pattern again... maybe see what it looks like as a while loop?

File: src/runtime/HalideBuffer.h (hunk 1 above)
Comment (steven-johnson) [reply src/runtime/HalideBuffer.h]: Done

File: src/runtime/cl_functions.h (hunk 2 above)
Comment (steven-johnson) [reply src/runtime/cl_functions.h]: Reverted whole file, marked with 'clang-format off'

File: src/runtime/cuda.cpp (hunk 3 above)
Comment (steven-johnson) [reply src/runtime/cuda.cpp]: this should be a switch statement instead, anyway

File: src/runtime/cuda.cpp (hunk 4 above)
Comment (steven-johnson) [reply src/runtime/cuda.cpp]: Moving the __attribute__ part to the end of the function make clang-format happier. (AFAICT it's legal in either position; correct me if I'm wrong.)

File: src/runtime/profiler.cpp (hunk 5 above)
Comment (steven-johnson) [reply src/runtime/profiler.cpp]: Done

File: src/runtime/profiler_inlined.cpp (hunk 6 above)
Comment (steven-johnson) [reply src/runtime/profiler_inlined.cpp]: `// clang-format off` FTW

File: src/runtime/tracing.cpp (hunk 7 above)
Comment (steven-johnson) [reply src/runtime/tracing.cpp]: done

File: src/runtime/write_debug_image.cpp (hunk 8 above)
Comment (steven-johnson) [reply src/runtime/write_debug_image.cpp]: DOne

File: src/runtime/write_debug_image.cpp (hunk 9 above)
Comment (steven-johnson) [reply src/runtime/write_debug_image.cpp]: done

**Comment by steven-johnson** [comment]
PTAL

**Pull request merged**